### PR TITLE
[9.0][FIX] account_invoice_merge: Invoices not related with sale order

### PR DESCRIPTION
**Pull request by carlosdauden**
self.env['sale.order'] is bolean evaluated as False

@Tecnativa

### CODE REVIEW

File: account_invoice_merge/models/account_invoice.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def make_key(br, fields):
         invoice_line_obj = self.env['account.invoice.line']
         # None if purchase is not installed
```
Comment (pedrobaeza) [inline account_invoice_merge/models/account_invoice.py]: Why this text?

Comment (carlosdauden) [reply account_invoice_merge/models/account_invoice.py]: Maybe we should ask @pedrobaeza @Cedric-Pigeon 
https://github.com/OCA/account-invoicing/blame/9.0/account_invoice_merge/models/account_invoice.py#L215
:wink:

Comment (pedrobaeza) [reply account_invoice_merge/models/account_invoice.py]: Well, with the blame, I only fix PEP8 issues, but the question is more as curiosity. Have you seen the PR status?

**Comment by LoisRForgeFlow** [comment]
@carlosdauden @pedrobaeza @sergio-teruel We have found this bug:
image
It pops after creating two invoices from the same sale order and trying to merge them.
image

**Comment by pedrobaeza** [comment]
@lreficent, as the later code was never run, there were no issues previously. The problem come in this code:
```
invoice_line_ids = invoice_line_obj.search(
    [('product_id', '=', so_line.product_id.id),
     ('invoice_id', '=', new_invoice_id)])
if invoice_line_ids:
    so_line.write({'invoice_lines': [(6, 0, invoice_line_ids)]})
```
It should be replaced by:
```
invoice_lines = invoice_line_obj.search(
    [('product_id', '=', so_line.product_id.id),
     ('invoice_id', '=', new_invoice_id)])
if invoice_lines:
    so_line.write({'invoice_lines': [(6, 0, invoice_lines.ids)]})
```
It's a bad conversion to new API.

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
https://github.com/OCA/account-invoicing/pull/262/files#diff-e3d50a35a4d448eec1952859dc589751R228 

 `{'invoice_lines': [(6, 0, invoice_line_ids)]})` -> `{'invoice_lines': [(6, 0, invoice_line_ids.ids)]})`

**Comment by LoisRForgeFlow** [comment]
@pedrobaeza yes, exactly. Can you add the fix here and we'll do in v10.

Also, we have found that [this line](https://github.com/OCA/account-invoicing/pull/262/files#diff-e3d50a35a4d448eec1952859dc589751R218) is not searching properly and gets all the sales orders.

**Comment by pedrobaeza** [comment]
@lreficent that can happen if `invoice_ids` is a computed field and there isn't a correct implementation of the search method. Can you check?

**Comment by LoisRForgeFlow** [comment]
@pedrobaeza I don't know if it is the cleanest way but it is fixed now in v10 (4e6e938e580a7d5619e3fcb6d9a116f6af08b771)

File: account_invoice_merge/models/account_invoice.py [hunk 2]
```diff
@@ -214,18 +214,18 @@ def make_key(br, fields):
...
         for new_invoice_id in invoices_info:
-            if so_obj:
-                todos = so_obj.search(
-                    [('invoice_ids', 'in', invoices_info[new_invoice_id])])
+            if so_obj != False:
```
Comment (MiquelRForgeFlow) [inline account_invoice_merge/models/account_invoice.py]: so_obj != False → so_obj is not False to make the travis works

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

**Comment by MiquelRForgeFlow** [comment]
Can you apply here the last fixes of https://github.com/OCA/account-invoicing/issues/238? 😊

**Comment by gurneyalex** [comment]
#269 fixes the build of the 10.0 branch

**Comment by pedrobaeza** [comment]
@carlosdauden this is waiting your final touch

**Review by MiquelRForgeFlow — APPROVED** [review]
:+1:

**Review by antoniocanovas — APPROVED** [review]
Ok, functionality ...

**Review by LoisRForgeFlow — APPROVED** [review]
Tested on runbot :+1:

**Pull request merged**